v2/examples/spiffe-jwt/README.md

```diff
@@ -0,0 +1,147 @@
+if err != nil {
+	log.Fatalf("Unable to connect to Workload API: %v", err)
+}
+svid, err := client.ValidateJWTSVID(ctx, token, audiences[0])
+```
+
+On the other side, the **HTTP client** uses the [workloadapi.X509Source](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/workloadapi?tab=doc#X509Source) to create a `tls.Config` for TLS that authenticates the server certificate and verifies that it has the SPIFFE ID `spiffe://examples.org/server`. 
+
+```go
+serverID := spiffeid.Must("example.org", "server")
+tlsConfig := tlsconfig.TLSClientConfig(source, tlsconfig.AuthorizeID(serverID))
+
+client := &http.Client{
+    Transport: &http.Transport{
+        TLSClientConfig: tlsConfig,
+    },
+}
+```
+
+Client fetches a JWT-SVID from the Workload API and adds it as a bearer token in the `Authorization` header.
+
+```go
+svid, err := jwtSource.FetchJWTSVID(ctx, jwtsvid.Params{
+	Audience: audience,
+})
+if err != nil {
+    log.Fatalf("Unable to fetch SVID: %v", err)
+}
+req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", svid.Marshal()))
+```
+
+That is it! The go-spiffe library fetches and automatically renews the X.509 SVID for the server and validates the client JWT SVIDs using the Workload API.
+
+As soon as the TLS connection is established, the client sends an HTTP request to the server and gets a response.
+
+## Building
+Build the client workload:
+```bash
+cd examples/spiffe-jwt/client
+go build
+```
+
+Build the server workload:
+```bash
+cd examples/spiffe-jwt/server
+go build
+```
+
+## Running
+This example assumes the following preconditions:
+- There is a SPIRE server and agent up and running.
+- There is a Unix workload attestor configured.
+- The trust domain is `example.org`
+- The agent SPIFFE ID is `spiffe://example.org/host`.
+- There is a `server-workload` and `client-workload` user in the system.
+
+### 1. Create the registration entries
+Create the registration entries for the client and server workloads:
+
+Server:
+```bash
+./spire-server entry create -spiffeID spiffe://example.org/server \
+                            -parentID spiffe://example.org/host \
+                            -selector unix:user:server-workload
+```
+
+Client: 
+```bash
+./spire-server entry create -spiffeID spiffe://example.org/client \
+                            -parentID spiffe://example.org/host \
+                            -selector unix:user:client-workload
+```
+
+### 2. Start the server
+Start the server with the `server-workload` user:
+```bash
+sudo -u server-workload ./server
+```
+
+### 3. Run the client
+Run the client with the `client-workload` user:
+```bash
+sudo -u client-workload ./client
+```
+
+The server should display a log `Request received` and client `Success!!!`
+
+To demonstrate a failure, an alternate audience value can be used. The server is expecting it's own SPIFFE ID as the audience value and will reject the token if it doesn't match.
+
+```
+sudo -u client-workload ./client spiffe://example.org/client
```

Comment:
Maybe we can use a "server-like" audience value. The client SPIFFE ID is a little confusing.
```suggestion
sudo -u client-workload ./client spiffe://example.org/some-other-server
```